Collect `PACKET_ID*` constants in `net_crypto.h`, cleanup their uses

oxij opened the pull request:
The result of me trying to understand how packet handling in tox works. This change is a noop, really.

I built `qtox`, `utox`, `toxic`, and `toxvpn` against this. All compile and, it seems, work as before.

<!-- Reviewable:start -->
---
This change is [<img src="https://reviewable.io/review_button.svg" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/toktok/c-toxcore/962)
<!-- Reviewable:end -->

oxij committed:
Collect `PACKET_ID*` constants in `net_crypto.h`, cleanup their uses

oxij committed:
Rename `m_handle_custom_lossy_packet -> m_handle_lossy_packet`

This function handles all lossy packets, including AV.

oxij committed:
Introduce several TODOs

CLAassistant commented:
[![CLA assistant check](https://cla-assistant.io/pull/badge/not_signed)](https://cla-assistant.io/TokTok/c-toxcore?pullRequest=962) <br/>Thank you for your submission! We really appreciate it. Like many open source projects, we ask that you sign our [Contributor License Agreement](https://cla-assistant.io/TokTok/c-toxcore?pullRequest=962) before we can accept your contribution.<br/><sub>You have signed the CLA already but the status is still pending? Let us [recheck](https://cla-assistant.io/check/TokTok/c-toxcore?pullRequest=962) it.</sub>

oxij commented:
@CLAassistant you're inaccessible through Tor (Cloudflare).

iphydf commented:
Thanks, never mind the CLA bot for now. I'll review this in a bit.

iphydf reviewed (changes requested):
> *[Reviewable](https://reviewable.io/reviews/toktok/c-toxcore/962)* status: 0 of 2 LGTMs obtained (waiting on @nurupo and @sudden6)

___
*[toxcore/BUILD.bazel, line 146 at r1](https://reviewable.io/reviews/toktok/c-toxcore/962#-LGisSxx9iRKaoEN9FqH:-LGisSxx9iRKaoEN9FqI:bqicoua) ([raw file](https://github.com/toktok/c-toxcore/blob/487a152d5223d72afc6fca5a49c95971014d12a4/toxcore/BUILD.bazel#L146)):*
> ```Python
>     name = "net_crypto",
>     srcs = ["net_crypto.c"],
>     hdrs = ["net_packet.h", "net_crypto.h"],
> ```

Run `buildifier`: `go get github.com/bazelbuild/buildtools/buildifier` and then `buildifier toxcore/BUILD.bazel`. Alternatively, make the list look like
```
    hdrs = [
        "net_crypto.h",
        "net_packet.h",
    ],
```

i.e. sorted, and one line per item.
___
*[toxcore/Messenger.h, line 180 at r1](https://reviewable.io/reviews/toktok/c-toxcore/962#-LGisoXZ5ev2HD8hlOEf:-LGisoXZ5ev2HD8hlOEg:b89w5nm) ([raw file](https://github.com/toktok/c-toxcore/blob/487a152d5223d72afc6fca5a49c95971014d12a4/toxcore/Messenger.h#L180)):*
> ```objc
> typedef int m_lossy_rtp_packet_cb(Messenger *m, uint32_t friendnumber, const uint8_t *data, uint16_t len, void *object);
> 
> struct RTP_Packet_Hadler {
> ```

s/Hadler/Handler/g
___
*[toxcore/Messenger.c, line 1816 at r1](https://reviewable.io/reviews/toktok/c-toxcore/962#-LGiswlL86uGAutz6CO3:-LGiswlL86uGAutz6CO4:bybvgty) ([raw file](https://github.com/toktok/c-toxcore/blob/487a152d5223d72afc6fca5a49c95971014d12a4/toxcore/Messenger.c#L1816)):*
> ```C
> 
>     if (packet[0] <= PACKET_ID_RANGE_LOSSY_AV_END) {
>         struct RTP_Packet_Hadler *ph =
> ```

`const` and `const` (pointer and pointee both).


<!-- Sent from Reviewable.io -->

oxij commented:
@iphydf Done, fixed, fixed, rebased/squashed, compiled, tested.

The original branch is `packet_h_orig`.

iphydf commented:
Something didn't seem to have worked. I'm still seeing the old code, no rebased code.

oxij commented:
@iphydf https://github.com/TokTok/c-toxcore/pull/962/files most certainly shows the new version.

iphydf commented:
@oxij ok, you're right. Can you rebase again? It's currently conflicting in Messenger.h.

oxij commented:
@iphydf Done. Everything still compiles and works.

robinlinden commented:
@oxij There are some style issues that need to be fixed before this passes Travis: https://travis-ci.org/TokTok/c-toxcore/jobs/403915817

iphydf commented:
I'm not really happy with the new header that comes without a .c file. Are you planning to factor out code from net_crypto.c into net_packet.c?

iphydf commented:
Why: because we usually have 1 module = 1 .c + 1 .h file.

oxij commented:
@robinlinden Fixed.

@iphydf

> Are you planning to factor out code from net_crypto.c into net_packet.c?

Not at the moment, maybe eventually.

> I'm not really happy with the new header that comes without a .c file.

I can merge it into `net_crypto.h` if you want, but I think it's much cleaner this way.

iphydf commented:
I'd prefer it merged, and build proper abstractions instead.

oxij commented:
Done.

iphydf reviewed (approved):
> *[Reviewable](https://reviewable.io/reviews/toktok/c-toxcore/962)* status: 0 of 1 LGTMs obtained (waiting on @nurupo)

___
*[toxcore/net_crypto.h, line 52 at r3](https://reviewable.io/reviews/toktok/c-toxcore/962#-LH_2dS_0xrzRRT15tYn:-LH_2dS_0xrzRRT15tYo:b-8tg94q) ([raw file](https://github.com/toktok/c-toxcore/blob/a21216644c0ada00c650b3e32529f84d01d8cbdd/toxcore/net_crypto.h#L52)):*
> ```objc
> #define PACKET_ID_RANGE_LOSSY_START 192
> #define PACKET_ID_RANGE_LOSSY_AV_START 192
> #define PACKET_ID_RANGE_LOSSY_AV_SIZE 8
> ```

Maybe it's better to define this number in terms of `END - START`.
___
*[toxcore/net_crypto.h, line 74 at r3](https://reviewable.io/reviews/toktok/c-toxcore/962#-LH_2wGA1RFeP0X9IvYs:-LH_2wGA1RFeP0X9IvYt:bhbyld3) ([raw file](https://github.com/toktok/c-toxcore/blob/a21216644c0ada00c650b3e32529f84d01d8cbdd/toxcore/net_crypto.h#L74)):*
> ```objc
> /* TODO(oxij): Make a sub-range instead?
>    But it's unclear what the sub-range end should be. 68? 79?
>    WTF is MSI packet anyway? It has a callback and no users.
> ```

MSI is for toxav.
___
*[toxcore/tox.c, line 1525 at r3](https://reviewable.io/reviews/toktok/c-toxcore/962#-LH_3G0D24lIPVC8TdbO:-LH_3G0D24lIPVC8TdbP:b1byfgs) ([raw file](https://github.com/toktok/c-toxcore/blob/a21216644c0ada00c650b3e32529f84d01d8cbdd/toxcore/tox.c#L1525)):*
> ```C
>     }
> 
>     // TODO(oxij): this is a hack, see commit introducing this line
> ```

Can you link to the commit? Why is it a hack?


<!-- Sent from Reviewable.io -->

iphydf reviewed (changes requested):
Reviewable made me approve, but I still have questions.

> *[Reviewable](https://reviewable.io/reviews/toktok/c-toxcore/962#-:-LH_3N9X1IxZBKETGuro:b-wbk2eo)* status: 0 of 1 LGTMs obtained (waiting on @nurupo)



<!-- Sent from Reviewable.io -->

iphydf commented:
Generally looks good. What do you think of the size vs. end-start idea?

oxij commented:
> Maybe it's better to define this number in terms of `END - START`.

That would be `END - START + 1`, which looks complicated, those lines will also get long. I like it the way it is.

> MSI is for toxav.

Ah, I see, I overlooked that. Removed that comment, added a comment to MSI message.

> Can you link to the commit? Why is it a hack?

Updated the comment.

iphydf reviewed (approved):
<img class="emoji" title=":lgtm_strong:" alt=":lgtm_strong:" align="absmiddle" src="https://reviewable.io/lgtm_strong.png" height="20" width="61"/>

> Reviewed 1 of 7 files at r2, 2 of 6 files at r3, 2 of 2 files at r4.
> *[Reviewable](https://reviewable.io/reviews/toktok/c-toxcore/962#-:-LHngekh4ATVYZcPDH8b:b-jbsf8z)* status: 0 of 1 LGTMs obtained (waiting on @nurupo)



<!-- Sent from Reviewable.io -->

iphydf commented:
I've rebased it, since you hadn't signed your commits, so I had to do that anyway.

oxij commented:
Ok. Thanks.
